### PR TITLE
bkz randomize makes sure that gso is updated (Fix #203)

### LINKED ISSUE #203: bkz.randomize_block seems broken
The permutation seems fine, but setting density>0 seems to have no effect

``` python
from fpylll import IntegerMatrix, LLL, FPLLL, GSO
from fpylll.algorithms.bkz2 import BKZReduction


A = IntegerMatrix(10, 10)
A.gen_identity(10)
M = GSO.Mat(A, float_type="double")
bkz = BKZReduction(M)

print(A)
print()
bkz.randomize_block(0, 10, density=5)

print(A)
print()
```
gives 

```
[ 1 0 0 0 0 0 0 0 0 0 ]
[ 0 1 0 0 0 0 0 0 0 0 ]
[ 0 0 1 0 0 0 0 0 0 0 ]
[ 0 0 0 1 0 0 0 0 0 0 ]
[ 0 0 0 0 1 0 0 0 0 0 ]
[ 0 0 0 0 0 1 0 0 0 0 ]
[ 0 0 0 0 0 0 1 0 0 0 ]
[ 0 0 0 0 0 0 0 1 0 0 ]
[ 0 0 0 0 0 0 0 0 1 0 ]
[ 0 0 0 0 0 0 0 0 0 1 ]

[ 0 0 0 0 0 0 1 0 0 0 ]
[ 0 1 0 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 0 1 ]
[ 0 0 0 0 1 0 0 0 0 0 ]
[ 1 0 0 0 0 0 0 0 0 0 ]
[ 0 0 0 1 0 0 0 0 0 0 ]
[ 0 0 1 0 0 0 0 0 0 0 ]
[ 0 0 0 0 0 1 0 0 0 0 ]
[ 0 0 0 0 0 0 0 0 1 0 ]
[ 0 0 0 0 0 0 0 1 0 0 ]


```

### PR DESCRIPTION
**Pull request by lducas**
A tentative fix for issue #203 .

### CODE REVIEW

File: src/fpylll/algorithms/bkz2.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def randomize_block(self, min_row, max_row, tracer=dummy_tracer, density=0):
         if max_row - min_row < 2:
             return  # there is nothing to do
 
+        for i in range(max_row):
+            self.M.update_gso_row(i, i)
+            print("X")
```
Comment (malb) [inline src/fpylll/algorithms/bkz2.py:49]: This seems like it's leftover from debugging

**Comment by malb** [comment]
So far, we don't do any `update_gso` calls for safety in any of the BKZ routines for performance reasons, so I'm not sure this is the right thing™ to do. At least we should measure the performance impact.

**Comment by lducas** [comment]
Ain't there a mechanism to recompute only what is actually not up to date ?

**Comment by malb** [comment]
`n_known_rows` is public so this might be the thing to branch on? We'd need to expose it to FPyLLL/Python first, though.

**Comment by lducas** [comment]
Ok, nvm.

**Pull request closed without merging**